src/libcore/python/xml_v.cpp

```diff
@@ -38,8 +47,160 @@ MTS_PY_EXPORT(xml) {
                     );
             }
             py::gil_scoped_release release;
-            return cast_object(
-                xml::load_string(name, mitsuba::detail::get_variant<Float, Spectrum>(), param));
+            return cast_object(xml::load_string(name, GET_VARIANT(), param));
         },
         "string"_a, D(xml, load_string));
+
+    m.def(
+        "load_dict",
+        [](const py::dict dict) {
+            return cast_object(load_dict<Float, Spectrum>(dict));
+        },
+        "dict"_a,
+R"doc(Load a Mitsuba scene or object from an Python dictionary
+
+Parameter ``dict``:
+    Python dictionary containing the object description
+
+)doc");
+}
+
+// Helper function to find a value of "type" in a Python dictionary
+std::string get_type(const py::dict &dict) {
+    for (auto& [key, value] : dict)
+        if (key.cast<std::string>() == "type")
+            return value.cast<std::string>();
+
+    Throw("Missing key 'type' in dictionary: %s", dict);
+}
+
+#define SET_PROPS(PyType, Type, Setter)         \
+    if (py::isinstance<PyType>(value)) {        \
+        props.Setter(key, value.cast<Type>());  \
+        continue;                               \
+    }
+
+MTS_VARIANT ref<Object> load_dict(const py::dict &dict) {
```

Comment:
Any specific use cases in mind? 

As you know, C++ isn't as easy with types compare to Python. Not sure it would be that easy to make it compatible with `std::map`.

Reply:
It could be useful to add a `create_object()` method to `Properties` directly, so we could build a `Properties` in python and later construct the corresponding object. What do you think?
This would mean exposing the `PluginManager::create_object()` function. Although we could have it live directly on `Properties` for simplicity.

Reply:
I think that even if we are able to construct objects from `Properties`, all of the work would still end up in this `load_dict` pybind function because we have to instantiate the children in order to create the `props` for the parents. Right?

Reply:
Indeed, you would have to instanciate the children one by one manually..